test/AutoRest.TestServer.Tests/CodeWriterTests.cs

```diff
@@ -78,7 +82,97 @@ public void CorrectlyHandlesCurlyBraces()
         {
             var codeWriter = new CodeWriter();
             codeWriter.Append($"public {typeof(string)} Data {{ get; private set; }}");
-            Assert.AreEqual("public string Data { get; private set; }" + Environment.NewLine, codeWriter.ToString(false));
+            Assert.AreEqual("public string Data { get; private set; }" + Environment.NewLine,
+                codeWriter.ToString(false));
+        }
+
+        [Test]
+        public void FormatInFormat()
+        {
+            var codeWriter = new CodeWriter();
+            FormattableString fs1 = $"'1' is {typeof(int)}";
+            FormattableString fs2 = $"'a' is {typeof(char)} and {fs1} and 'true' is {typeof(bool)}";
+
+            codeWriter.Append(fs2);
+            var expected = "'a' is char and '1' is int and 'true' is bool" + Environment.NewLine;
+            Assert.AreEqual(expected, codeWriter.ToString(false));
+        }
+
+
+        [Test]
+        public void EnumerableFormatInFormat()
+        {
+            var codeWriter = new CodeWriter();
+            codeWriter.Append($"Multiply:{Enumerable.Range(1, 4).Select(i => (FormattableString)$" {i} * 2 = {i * 2};")}");
+            var expected = "Multiply: 1 * 2 = 2; 2 * 2 = 4; 3 * 2 = 6; 4 * 2 = 8;" + Environment.NewLine;
+            Assert.AreEqual(expected, codeWriter.ToString(false));
+        }
+
+        [Test]
+        public void SingleLineSummary()
+        {
+            var codeWriter = new CodeWriter();
+            codeWriter.WriteXmlDocumentationSummary($"Some {typeof(string)} summary.");
+            var expected = "/// <summary> Some string summary. </summary>" + Environment.NewLine;
+            Assert.AreEqual(expected, codeWriter.ToString(false));
+        }
+
+        [Test]
+        public void NoEmptySummary()
+        {
+            var codeWriter = new CodeWriter();
+            codeWriter.WriteXmlDocumentationSummary($"{string.Empty}");
+            var expected = string.Empty;
+            Assert.AreEqual(expected, codeWriter.ToString(false));
+        }
+
+        [Test]
+        public void MultiLineSummary()
```

Comment:
Can you please add a few tests around start/end tag logic in AppendXmlDocumentation